src/lib/utils/hooks/useOverlay.tsx

```diff
@@ -1,63 +1,104 @@
-import React, { RefObject, useRef, useState } from 'react'
+import React, {
+  RefObject,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
 import { Overlay } from 'react-bootstrap'
 import { Placement } from 'react-bootstrap/esm/Overlay'
 
-function resetTimer(timer: React.MutableRefObject<NodeJS.Timeout | null>) {
-  if (timer.current) {
-    clearTimeout(timer.current)
+function resetTimer(timer: NodeJS.Timeout | null) {
+  if (timer) {
+    clearTimeout(timer)
   }
 }
 
+const DEFAULT_DELAY_SHOW_MS = 250
+const DEFAULT_DELAY_HIDE_MS = 500
+
 export function useOverlay(
```

Comment:
For some reason, this re-renders like crazy for no reason. Doesn't seem to cause performance issues, but makes it hard to test. I added a bunch of `useCallback`s and now keep track of the mounted state to ensure we don't update state after unmounting. No major logic changes.